dot/types/consensus_digest.go

```diff
@@ -28,12 +28,12 @@ type GrandpaScheduledChange struct {
 
 // Encode returns a SCALE encoded GrandpaScheduledChange with first type byte
 func (sc *GrandpaScheduledChange) Encode() ([]byte, error) {
```

Comment:
please take this refactor a step further and remove the `Digest` interface.  I don't think we need it, since the interface is just handling the encoding of the type byte, and the actual scale encoding.

Reply:
Okay sounds good. With the changes from the above comment I should be able to get rid of it

Reply:
@timwu20 Does the types/consensus_digest refactor look alright to you?

Reply:
You don't need all the const types anymore, if you have them defined in the `Index()` receiver func that fulfills the `VaryingDataTypeValue` interface.

Reply:
@timwu20 okay I removed the const types. That look okay to you?

Reply:
The only thing I would mention is maybe we shouldn't use the same variables for the VDTs.

```
var BabeConsensusDigest = scale.MustNewVaryingDataType(NextEpochData{}, BABEOnDisabled{}, NextConfigData{})
var GrandpaConsensusDigest = scale.MustNewVaryingDataType(GrandpaScheduledChange{}, GrandpaForcedChange{}, GrandpaOnDisabled{}, GrandpaPause{}, GrandpaResume{})
```

This isn't thread safe, and could product nasty side effects if multiple goroutines are trying to unmarshal.  Maybe create a constructor method that returns you a varying data type.  

```
func NewBabeConsensusDigest() scale.VaryingDataType {
   return scale.MustNewVaryingDataType(NextEpochData{}, BABEOnDisabled{}, NextConfigData{})
}
```

This will set us up for when I support the custom `scale.VaryingDataType`.

Reply:
Good call. Will change this